### PR TITLE
bump elasticsearch version

### LINKED ISSUE #3748: Upgrade to elasticsearch 6
We should do #2867 first, I'm not sure if skipping major versions is a good idea

### PR DESCRIPTION
**Pull request by tasawernawaz**
#### What are the relevant tickets?
Fixes #3748 

#### What's this PR do?
Bump elastic search version to 6.

#### How should this be manually tested?
Search should work in the app.

### CODE REVIEW

**Comment by tasawernawaz** [comment]
@gsidebo After bumping up `elasticsearch` version, few tests were failing. I have tried to fix them, still couple of tests are failing. Could you please look at those and review other changes.

**Comment by tasawernawaz** [comment]
@noisecapella I have used `search` instead of `percolate` method as they have removed it in the favor of `search`. Details at https://github.com/elastic/elasticsearch/pull/22331/files#diff-9c538c3ce0976b32ffd81a4ffeb389e4R6

Could you please do a review as build has passed now.

File: search/indexing_api.py [hunk 1]
```diff
@@ -230,6 +231,8 @@
     }
 }
 
+TEST_INDEX_WILDCARD = '{index_name}_*'.format(index_name=settings.ELASTICSEARCH_INDEX)
```
[the remark this answers is not in the document]
Comment (tasawernawaz) [reply search/indexing_api.py]: Done.
`_all` we override sometimes `settings.ELASTICSEARCH_INDEX` in tests, so don't want to risk it delete all indices.

**Comment by tasawernawaz** [comment]
@pdpinch @mitodl/devops is there any pre/post requisite to the release of this upgrade ?

**Comment by blarghmatey** [comment]
Yes, we will need to upgrade the ES instances, so we will coordinate with the release master for this.

**Pull request merged**